### PR TITLE
4 exercises done

### PR DESCRIPTION
**Pull request by Romchik38**

### CODE REVIEW

File: Exercises/3-by-reference.js [hunk 1]
```diff
@@ -1,5 +1,9 @@
 'use strict';
 
-const inc = null;
-
+const a = { n: 5 };
+const inc = num => {
+  ++num.n;
+};
+inc(a);
+console.log(a);
```
Comment (tshemsedinov) [inline Exercises/3-by-reference.js]: Remove debug output

File: Exercises/4-count-types.js [hunk 2]
```diff
@@ -1,5 +1,43 @@
...
+  const obj = {};
+  for (const v of arr) {
+    obj[typeof v] === undefined ?
+      obj[typeof v] = 1 :
+      ++obj[typeof v];
+  }
+  return obj;
+};
+
+console.log(countTypesInArray(arr));
```
Comment (tshemsedinov) [inline Exercises/4-count-types.js]: Remove debug output

Comment (tshemsedinov) [reply Exercises/4-count-types.js]: Put repeating expressions to local identifiers, for example: `typeof v` and `obj[typeof v]`

File: Exercises/3-by-reference.js (hunk 1 above)
Comment (Romchik38) [reply Exercises/3-by-reference.js]: done

File: Exercises/4-count-types.js (hunk 2 above)
Comment (Romchik38) [reply Exercises/4-count-types.js]: done

File: Exercises/1-hoisting.js [hunk 3]
```diff
@@ -1,5 +1,8 @@
 'use strict';
 
-const fn = null;
+const fn = function () {
```
Comment (tshemsedinov) [inline Exercises/1-hoisting.js]:
```suggestion
const fn = () => {
```

File: Exercises/3-by-reference.js [hunk 4]
```diff
@@ -1,5 +1,8 @@
 'use strict';
 
-const inc = null;
-
+const a = { n: 5 };
```
Comment (tshemsedinov) [inline Exercises/3-by-reference.js]: Remove this use case
```suggestion
```

File: Exercises/3-by-reference.js [hunk 5]
```diff
@@ -1,5 +1,8 @@
...
+const inc = num => {
+  ++num.n;
+};
+inc(a);
```
Comment (tshemsedinov) [inline Exercises/3-by-reference.js]: Remove this call, it's just for debug. Tests will do that for you.
```suggestion
```

File: Exercises/4-count-types.js [hunk 6]
```diff
@@ -1,5 +1,20 @@
...
+  1, 'Kiev', true, -100, undefined,
+  false, { name: 'ser' }, 20, 'Warrior life', true
+];
+
+const countTypesInArray = arr => {
+  const obj = {};
+  for (const v of arr) {
+    const type = typeof v;
+    const count = obj[type];
+    count === undefined ?
```
Comment (tshemsedinov) [inline Exercises/4-count-types.js]: Ternary is not good choice here, try to improve readability.

**Pull request closed without merging**